addons/l10n_es_pos/static/src/overrides/components/payment_screen/PaymentScreen.js

```diff
@@ -0,0 +1,24 @@
+/** @odoo-module */
+import { patch } from "@web/core/utils/patch";
+import { PaymentScreen } from "@point_of_sale/app/screens/payment_screen/payment_screen";
+
+patch(PaymentScreen.prototype, "l10n_es_pos.PaymentScreen", {
+    async validateOrder(isForceValidate) {
+        const _super = this._super;
+        if (!this.pos.config.is_spanish) {
+            await _super(...arguments);
+            return;
+        }
+        const below_limit =
+            this.currentOrder.get_total_with_tax() <=
+            this.pos.config.l10n_es_simplified_invoice_limit;
+        const order = this.currentOrder;
+        if (below_limit && !order.to_invoice) {
+            await order.set_simple_inv_number();
```

Comment:
i had the exact though. I figured that since it is working well and it is also quite readable it didn't make sense to change it. 
I'll change it if you think that it's worth it

Reply:
If you tested it offline, with slow 3g and with normal internet connection and it is working the way we want I'm ok if it stays like this

Reply:
in the end i refactored using `async` `await`